src/lib/download-manager.js

```diff
@@ -96,6 +98,73 @@ class DownloadManager {
 		}
 	}
 
+	async _validateCacheLimit({ url, currentFileName, alwaysCleanCache = false }) {
+		const maxCacheSizeGB = parseFloat(settings.tachyonCacheLimitGB);
+		const fileHeaders = await fetch(url, { method: 'HEAD' });
+
+		const contentLength = parseInt(fileHeaders.headers.get('content-length') || '0', 10);
+		// get the size of the download directory
+		const downloadDirStats = await this.getDownloadedFilesStats(currentFileName);
+		const totalSizeGB = (downloadDirStats.totalSize + contentLength) / (1024 * 1024 * 1024); // convert to GB
+		const downloadedCacheSizeGB = downloadDirStats.totalSize / (1024 * 1024 * 1024); // convert to GB
+		const shouldCleanCache = await this._shouldCleanCache({
+			downloadedCacheSizeGB,
+			alwaysCleanCache,
+			maxCacheSizeGB,
+			totalSizeGB,
+			downloadDirStats
+		});
+		if (shouldCleanCache) {
+			await this._freeCacheSpace(downloadDirStats);
+		}
+		return;
+	}
+
+	async getDownloadedFilesStats(currentFile) {
+		const files = await fs.readdir(this.downloadDir);
+		// use just the zip files and progress files
+		const packageFiles = files.filter(file => file.endsWith('.zip') || file.endsWith('.progress'));
+		// exclude the current file from the list
+		const filteredFiles = currentFile ? packageFiles.filter(file => file !== currentFile && file !== `${currentFile}.progress`) : packageFiles;
+		// stat to get size, and mtime to sort by date modified
+		const fileStats = await Promise.all(filteredFiles.map(async (file) => {
+			const filePath = path.join(this.downloadDir, file);
+			const stats = await fs.stat(filePath);
+			return { filePath, size: stats.size, mtime: stats.mtime };
+		}));
+		// sort files by date modified
+		const sortedFileStats = fileStats.sort((a, b) => a.mtime - b.mtime);
+
+		return {
+			totalSize: sortedFileStats.reduce((sum, file) => sum + file.size, 0),
+			fileStats
+		};
+	}
+
+	async _shouldCleanCache({ downloadedCacheSizeGB, alwaysCleanCache, maxCacheSizeGB, totalSizeGB, downloadDirStats }) {
+		if (parseInt(maxCacheSizeGB) === 0 || alwaysCleanCache) {
```

Comment:
parseInt shouldn't be needed anymore here because it's done where you set `maxCacheSizeGB = parseFloat(settings.tachyonCacheLimitGB)`, right?

Reply:
Right! Fixed here! https://github.com/particle-iot/particle-cli/pull/789/commits/e34514f263c69bd0d9e8a44df21ac44fa097235c